conans/client/shims/write_shims.py

```diff
@@ -0,0 +1,66 @@
+import os
+import stat
+import textwrap
+
+from jinja2 import Template
+
+from conans.client.envvars.environment import BAT_FLAVOR, SH_FLAVOR, env_files
+from conans.client.run_environment import RunEnvironment
+from conans.util.files import save_files
+
+cmd_template = textwrap.dedent("""\
+    echo Calling {{ name }} wrapper
+    call "{{ activate_path }}"
+    pushd "{{ exe_path_dirname }}"
+    call "{{ exe_path }}" %*
+    popd
+    call "{{ deactivate_path }}"
+    """)
+
+sh_template = textwrap.dedent("""\
+    #!/bin/bash
+    echo Calling {{ name }} wrapper
+    source "{{ activate_path }}"
+    pushd "{{ exe_path_dirname }}" > /dev/null
+    "{{ exe_path }}" "$@"
+    popd > /dev/null
+    source "{{ deactivate_path }}"
+    """)
+
+
+def _generate_shim(name, conanfile, settings_os, output_path):
+    # Use the environment generators we already have
+    suffix = "_{}".format(name)
+    environment = RunEnvironment(conanfile).vars
+    for k, v in conanfile.env_info.vars.items():
+        environment[k] = v  # TODO: Append to known variables
```

Comment:
This behavior should be available somewhere else: here I need the `RunEnvironment` but also the environment defined in the recipe itself (`self.env_info`).